api/agent/call.go

```diff
@@ -325,27 +324,15 @@ func (c *call) Start(ctx context.Context) error {
 	c.StartedAt = common.DateTime(time.Now())
 	c.Status = "running"
 
-	if c.Type == models.TypeAsync {
-		// XXX (reed): make sure MQ reservation is lengthy. to skirt MQ semantics,
-		// we could add a new message to MQ w/ delay of call.Timeout and delete the
-		// old one (in that order), after marking the call as running in the db
-		// (see below)
-
-		// XXX (reed): should we store the updated started_at + status? we could
-		// use this so that if we pick up a call from mq and find its status is
-		// running to avoid running the call twice and potentially mark it as
-		// errored (built in long running task detector, so to speak...)
-
-		err := c.handler.Start(ctx, c.Model())
-		if err != nil {
-			return err // let another thread try this
-		}
+	err := c.handler.Start(ctx, c.Model())
```

Comment:
this will still make an api call if the hybrid api is being used, not sure it's what we want

Reply:
can we gate the hybrid client not to make this request if async? we also need to change direct data access to not delete the mq message here if it's sync, that call will fail too (oddly tests don't fail, this worries me)

Reply:
https://github.com/fnproject/fn/blob/master/api/agent/data_access.go#L157 think we need a similar gate here around if async